autoload/neomake.vim

```diff
@@ -236,9 +243,9 @@ function! s:AddExprCallback(maker) abort
     let place_signs = get(g:, 'neomake_place_signs', 1)
     let list = file_mode ? getloclist(a:maker.winnr) : getqflist()
 
-    while s:neomake_list_nr < len(list)
-        let entry = list[s:neomake_list_nr]
-        let s:neomake_list_nr += 1
+    while b:neomake_list_nr < len(list)
```

Comment:
In order for the `b:` prefix to work, we will need to be sure that this function can never be called when the user is not in the buffer that initiated `:Neomake`. I'm not sure if that's the case right now.